pkg/artifactory/provider.go

```diff
@@ -69,64 +69,82 @@ func Provider() *schema.Provider {
-			"artifactory_local_go_repository":        resourceArtifactoryLocalGenericRepository("go"),
-			"artifactory_local_conan_repository":     resourceArtifactoryLocalGenericRepository("conan"),
-			"artifactory_local_chef_repository":      resourceArtifactoryLocalGenericRepository("chef"),
-			"artifactory_local_puppet_repository":    resourceArtifactoryLocalGenericRepository("puppet"),
-			"artifactory_local_maven_repository":     resourceArtifactoryLocalJavaRepository("maven", false),
-			"artifactory_local_gradle_repository":    resourceArtifactoryLocalJavaRepository("gradle", true),
-			"artifactory_local_alpine_repository":    resourceArtifactoryLocalAlpineRepository(),
-			"artifactory_local_debian_repository":    resourceArtifactoryLocalDebianRepository(),
-			"artifactory_local_docker_v2_repository": resourceArtifactoryLocalDockerV2Repository(),
-			"artifactory_local_docker_v1_repository": resourceArtifactoryLocalDockerV1Repository(),
-			"artifactory_local_rpm_repository":       resourceArtifactoryLocalRpmRepository(),
-			"artifactory_remote_repository":          resourceArtifactoryRemoteRepository(),
-			"artifactory_remote_npm_repository":      resourceArtifactoryRemoteNpmRepository(),
-			"artifactory_remote_docker_repository":   resourceArtifactoryRemoteDockerRepository(),
-			"artifactory_remote_helm_repository":     resourceArtifactoryRemoteHelmRepository(),
-			"artifactory_remote_cargo_repository":    resourceArtifactoryRemoteCargoRepository(),
-			"artifactory_virtual_repository":         resourceArtifactoryVirtualRepository(),
-			"artifactory_virtual_maven_repository":   resourceArtifactoryMavenVirtualRepository(),
-			"artifactory_virtual_go_repository":      resourceArtifactoryGoVirtualRepository(),
-			"artifactory_virtual_conan_repository":   resourceArtifactoryVirtualRepositoryWithRetrievalCachePeriodSecs("conan"),
-			"artifactory_virtual_generic_repository": resourceArtifactoryVirtualGenericRepository("generic"),
-			"artifactory_group":                      resourceArtifactoryGroup(),
-			"artifactory_user":                       resourceArtifactoryUser(),
-			"artifactory_permission_target":          resourceArtifactoryPermissionTarget(),
-			"artifactory_pull_replication":           resourceArtifactoryPullReplication(),
-			"artifactory_push_replication":           resourceArtifactoryPushReplication(),
-			"artifactory_certificate":                resourceArtifactoryCertificate(),
-			"artifactory_api_key":                    resourceArtifactoryApiKey(),
-			"artifactory_access_token":               resourceArtifactoryAccessToken(),
-			"artifactory_general_security":           resourceArtifactoryGeneralSecurity(),
-			"artifactory_oauth_settings":             resourceArtifactoryOauthSettings(),
-			"artifactory_saml_settings":              resourceArtifactorySamlSettings(),
+			"artifactory_keypair":                        resourceArtifactoryKeyPair(),
```

Comment:
This map is actually getting kind of silly. Would it be too much to ask for a loop insertion? Because this will only get worse